src/index.spec.ts

```diff
@@ -1310,6 +1312,27 @@ describe('EppoPrecomputedJSClient E2E test', () => {
       format: 'PRECOMPUTED',
     });
   });
+
+  it('deduplicates assignment logging', () => {
+    // Reset the mock logger and assignment cache before this test
+    const mockLogger = td.object<IAssignmentLogger>();
+    const mockAssignmentCache = td.object<AssignmentCache>();
+    td.when(mockAssignmentCache.has(td.matchers.anything())).thenReturn(false, true);
```

Comment:
I needed to reset the assignment cache at the top of this test, and mocking its behavior is one way to do it. I don't like this approach though because telling the function when to return false or true is removing the part of the test that ensures that the assignment cache is setting assignments correctly

Some alternative approaches

Use the same code as the precomputedInit method to initialize a new hybrid cache

https://github.com/Eppo-exp/js-client-sdk/blob/56c4555e7a2ac1b11257eb6297e79c9eca4a1c0a/src/index.ts#L697-L705

Or switch to using an in-memory cache. The methods for enabling this are convenient

```
globalClient.useNonExpiringInMemoryAssignmentCache()
```